dom.go

```diff
@@ -418,20 +454,26 @@ func renderComponent(comp Component, prevRender *HTML) (h *HTML, skip bool) {
 		nextRender = Tag("noscript")
 	}
 
-	// Restore the actual rendered HTML.
+	// Reconcile the actual rendered HTML.
 	nextRender.reconcile(prevRender)
 
 	// Update the context to consider this render.
-	comp.Context().prevRender = nextRender
-	comp.Context().prevComponent = comp
-	comp.Context().prevRenderComponent = doCopy(comp)
+	//
+	// TODO: In the event that this really is the previous component instance,
```

Comment:
Actually I don't think that is it that. The bug here is that:

1. Imagine a two-component setup: a parent, and its child.
1. `Rerender(thePrevParent)` is called, where `thePrevParent` is the 'previous' component instance (one the user has a pointer to).
1. `renderComponent(thePrevParent, ...)` will be called and so the 'previous' component instance (`thePrevParent.Context()`) will be updated on lines 467-469 below.
1. When its children are walked, it will effectively invoke `renderComponent(theNextChild, thePrevChildContext, thePrevChildRender)`, with `theNextChild` being the newly created component instance -- NOT the persistent / previous component instance. So the 'next' component instance (`theNextChild.Context()`) will be updated on lines 467-469.

To be entirely honest, this is hard for me to reason about, and that's part of the problem^. The troublesome bit here is that... it's not clear whether or not the 'next' or 'previous' `Component.Context` has the information and which needs to be updated.

I'll play around with this some more to see if I can resolve this issue / remove the ambiguity..